### PR TITLE
BNGP-2885- initial commit for terms and conditions

### PR DESCRIPTION
**Pull request by usman-j-shaikh**

### CODE REVIEW

**Review by paulsimonandrews — APPROVED** [review]
LGTM

File: packages/webapp/src/utils/credits-purchase-constants.js [hunk 1]
```diff
@@ -1,12 +1,19 @@
 const routes = {
-  ESTIMATOR_CREDITS_INDIVIDUAL_ORG: '/credits-purchase/applying-individual-organisation'
+  ESTIMATOR_CREDITS_INDIVIDUAL_ORG: '/credits-purchase/applying-individual-organisation',
+  CREDITS_TERM_AND_CONDITIONS: '/credits-purchase/credits-terms-and-conditions',
```
Comment (paulsimonandrews) [inline packages/webapp/src/utils/credits-purchase-constants.js:3]: Use `CREDITS_PURCHASE_TERMS_AND_CONDITIONS`

Also the URL should be `/credits-purchase/confirm-terms-conditions`

File: packages/webapp/src/utils/credits-purchase-constants.js [hunk 2]
```diff
@@ -1,12 +1,19 @@
...
+  CREDITS_CHECK_YOUR_ANSWERS: '/credits-purchase/credits-check-your-answers'
```
Comment (paulsimonandrews) [inline packages/webapp/src/utils/credits-purchase-constants.js:4]: Use `CREDITS_PURCHASE_CHECK_YOUR_ANSWERS`

File: packages/webapp/src/utils/credits-purchase-constants.js [hunk 3]
```diff
@@ -1,12 +1,19 @@
...
 }
 
 const views = Object.fromEntries(
   Object.entries(routes).map(([k, v]) => [k, v.substring(1)])
 )
 
+const redisKeys = {
+  CREDITS_TERMS_AND_CONDITIONS: 'credits-purchase-consent'
```
Comment (paulsimonandrews) [inline packages/webapp/src/utils/credits-purchase-constants.js:12]: Use `CREDITS_PURCHASE_TERMS_AND_CONDITIONS_CONFIRMED`

File: packages/webapp/src/views/credits-purchase/credits-terms-and-conditions.html [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        {% if err %}
+            {% include "errorbanner.html" %}
+        {% endif %}
+        <div class="govuk-grid-row">
+            <div class="govuk-grid-column-two-thirds">
+              <h1 class="govuk-heading-xl">{{ pageHeading }}</h1>
+                <p>The statutory biodiversity credit scheme is run by Natural England on behalf of the Department for Environment, Food and Rural Affairs (Defra) as part of the Environment Act 2021.</p>
+                
+                <h2 class="govuk-heading-m">Privacy notice</h2>
+                <p class="govuk-body">Our <a href="#">privacy notice </a>sets out how we use the personal data.</p>
```
Comment (paulsimonandrews) [inline packages/webapp/src/views/credits-purchase/credits-terms-and-conditions.html:17]: Use the same link as we use in the footer to the privacy notice: https://www.gov.uk/government/publications/natural-england-privacy-notices/biodiversity-net-gain-privacy-notice

File: packages/webapp/src/views/credits-purchase/credits-terms-and-conditions.html [hunk 5]
```diff
@@ -0,0 +1,58 @@
+{% extends 'form-layout.html' %}
+
+{% set serviceName = "Purchase statutory biodiversity credits" %}
+{% set pageHeading = "Terms and Conditions for the purchase of Statutory Biodiversity Credits" %}
```
Comment (paulsimonandrews) [inline packages/webapp/src/views/credits-purchase/credits-terms-and-conditions.html:4]: Only `Terms` should be capitalised, the rest should all be lowercase

File: packages/webapp/src/views/credits-purchase/credits-terms-and-conditions.html [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+{% block formContent %}
+    <div class="govuk-body">
+        {% if err %}
+            {% include "errorbanner.html" %}
+        {% endif %}
+        <div class="govuk-grid-row">
+            <div class="govuk-grid-column-two-thirds">
+              <h1 class="govuk-heading-xl">{{ pageHeading }}</h1>
```
Comment (paulsimonandrews) [inline packages/webapp/src/views/credits-purchase/credits-terms-and-conditions.html:13]: Missing an `<h2>Introduction</h2>` that is in the copydeck

File: packages/webapp/src/views/credits-purchase/credits-terms-and-conditions.html [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+                <h2 class="govuk-heading-m">Privacy notice</h2>
+                <p class="govuk-body">Our <a href="#">privacy notice </a>sets out how we use the personal data.</p>  
+                
+                <h2 class="govuk-heading-m">Your responsibilities</h2>
+                <p class="govuk-body">It is your responsibility to ensure:</p>
+                <ul class="govuk-list govuk-list--bullet">
+                  <li>you have purchased the correct number, and tier, of statutory biodiversity credits to meet your BNG obligation for development under the Planning Permission referenced in your application</li>
+                </ul>
+                
+                <h2 class="govuk-heading-m">Anti-money laundering</h2>
```
Comment (paulsimonandrews) [inline packages/webapp/src/views/credits-purchase/credits-terms-and-conditions.html:25]: Heading should be: Customer due diligence

File: packages/webapp/src/views/credits-purchase/credits-terms-and-conditions.html [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+                <p class="govuk-body">All applications to purchase statutory biodiversity credits are subject to anti-money laundering checks. They can only be purchased by applicants who pass those checks.</p>
```
Comment (paulsimonandrews) [inline packages/webapp/src/views/credits-purchase/credits-terms-and-conditions.html:26]: Should be: "All applications to purchase statutory biodiversity credits are subject to customer due diligence (anti-money laundering) checks. They can only be purchased by applicants who pass those checks."

File: packages/webapp/src/views/credits-purchase/credits-terms-and-conditions.html [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+                
+                <h2 class="govuk-heading-m">Purchase terms</h2>
+                <p class="govuk-body">You have no obligation to Natural England to buy the statutory biodiversity credits in this service. The terms and conditions for purchase of statutory biodiversity credits will apply if you proceed.</p>
+                
+                <h2 class="govuk-heading-m">Introduction</h2>
```
Comment (paulsimonandrews) [inline packages/webapp/src/views/credits-purchase/credits-terms-and-conditions.html:31]: This heading should be at the top, and the paragraph below is a repeat of what is at the top

File: packages/webapp/src/views/credits-purchase/credits-terms-and-conditions.html [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+                
+
+                {{ govukCheckboxes({
+                  name: "termsAndConditions",
+                  class: "govuk-grid-column-two-thirds",
+                  errorMessage: err[0],
+                  items: [
+                    {
+                      value: "true",
+                      html: "I confirm I have read the terms and conditions."
```
Comment (paulsimonandrews) [inline packages/webapp/src/views/credits-purchase/credits-terms-and-conditions.html:44]: Should be "I confirm I have read the above, and the full terms and conditions"

File: packages/webapp/src/views/credits-purchase/credits-terms-and-conditions.html [hunk 11]
```diff
@@ -0,0 +1,58 @@
+{% extends 'form-layout.html' %}
+
+{% set serviceName = "Purchase statutory biodiversity credits" %}
```
Comment (paulsimonandrews) [inline packages/webapp/src/views/credits-purchase/credits-terms-and-conditions.html:3]: Should be: "Buy statutory biodiversity credits"

File: packages/webapp/src/routes/credits-purchase/credits-terms-and-conditions.js [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+const handlers = {
+  get: async (request, h) => {
+    return h.view(creditsPurchaseConstants.views.CREDITS_TERM_AND_CONDITIONS)
+  },
+  post: async (request, h) => {
+    const consent = request.payload.termsAndConditions
+    if (!consent) {
+      return h.view(creditsPurchaseConstants.views.CREDITS_TERM_AND_CONDITIONS, {
+        err: [{
+          text: 'Check the box to confirm you have read the Ts and Cs',
```
Comment (paulsimonandrews) [inline packages/webapp/src/routes/credits-purchase/credits-terms-and-conditions.js:12]: Should be "Check the box to confirm you have read the terms and conditions"

File: packages/webapp/src/routes/__tests__/credits-purchase/credits-terms-and-conditions.spec.js [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+    it('Should continue journey to consent being ticked', async () => {
+      postOptions.payload.termsAndConditions = 'true'
+      const res = await submitPostRequest(postOptions)
+      expect(res.headers.location).toEqual(creditsPurchaseConstants.routes.CREDITS_CHECK_YOUR_ANSWERS)
+    })
+    it('Should stop journey if consent not ticked', async () => {
+      postOptions.payload.termsAndConditions = undefined
+      const res = await submitPostRequest(postOptions, 200)
+      expect(res.payload).toContain('There is a problem')
+      expect(res.payload).toContain('Check the box to confirm you have read the Ts and Cs')
```
Comment (paulsimonandrews) [inline packages/webapp/src/routes/__tests__/credits-purchase/credits-terms-and-conditions.spec.js:28]: Will need to change as content is wrong in the page

**Review by paulsimonandrews — CHANGES_REQUESTED** [review]
Please update constant names and content errors as requested in comments

File: packages/webapp/src/utils/credits-purchase-constants.js [hunk 14]
```diff
@@ -1,13 +1,17 @@
 const routes = {
-  CREDITS_PURCHASE_CHECK_DEFRA_ACCOUNT_DETAILS: '/credits-purchase/check-defra-account-details'
+  CREDITS_PURCHASE_CHECK_DEFRA_ACCOUNT_DETAILS: '/credits-purchase/check-defra-account-details',
+  CREDITS_PURCHASE_INDIVIDUAL_ORG: '/credits-purchase/applying-individual-organisation',
+  CREDITS_PURCHASE_TERMS_AND_CONDITIONS: '/credits-purchase/credits-terms-and-conditions',
```
Comment (paulsimonandrews) [inline packages/webapp/src/utils/credits-purchase-constants.js:4]: URL should be: `/credits-purchase/confirm-terms-conditions`

**Review by paulsimonandrews — CHANGES_REQUESTED** [review]
Just need to change the URL for the route (and rename the route and views files accordingly)

**Review by paulsimonandrews — APPROVED** [review]

**Review by paulsimonandrews — APPROVED** [review]
LGTM

**Comment by shaundefra** [comment]
Tested by Tajinder

**Pull request merged**